### PR TITLE
Add notes and research on error budgets

### PR DESCRIPTION
**Pull request by csomh**
Signed-off-by: Hunor Csomortáni <csomh@redhat.com>

### CODE REVIEW

File: error-budgets/README.md [hunk 1]
```diff
@@ -0,0 +1,319 @@
...
+reproducible way.
+
+##### Forming an Error Budget
+
+Based on the service level objective (SLO).
+
+Defines a clear, objective metric that determines how unreliable a service is
+allowed to be within a quarter.
+
+1. Product Management defines the SLO, which sets and expectation for the
```
Comment (TomasTomecek) [inline error-budgets/README.md:161]:
```suggestion
1. Product Management defines the SLO, which sets an expectation for the
```

File: error-budgets/README.md [hunk 2]
```diff
@@ -0,0 +1,319 @@
...
+   uptime.
+2. Monitoring system (neutral third party) measures the uptime.
+3. The previous two define the "budget" of how much "unreliability" is left in
+   the quarter.
+4. As long as there is error budget remaining, releases can be pushed.
+
+##### Benefits
+
+Discussions/decisions around release velocity can be driven by the error
+budget.
```
Comment (TomasTomecek) [inline error-budgets/README.md:171]: Wow, so we could actually deploy to prod multiple times a week :O

**Review by TomasTomecek — APPROVED** [review]
I already have questions for the SLIs and SLOs and will hold my horses for the discussion with stakeholders.

Since you pointed out so many high-volume projects - would a survey be an efficient way of gathering feedback?

File: error-budgets/README.md [hunk 3]
```diff
@@ -0,0 +1,319 @@
...
+3. Figure out which of the expectations identified in the previous step can be
+   measured with an indicator. What are these indicators (SLIs)? What are the
+   SLOs?
+   - We might want to check whether the SLOs make sense to the stakeholders.
+4. Capture/produce the indicators (metrics/monitoring)
+5. Track indicators to know how we are doing against our targets (SLOs). This
+   means setting up monitoring dashboards and alerts, and making sure we have
+   the process in place to become aware and take action (if needed) in case
+   objectives are in danger to be missed.
+
```
[the remark this answers is not in the document]
Comment (csomh) [reply error-budgets/README.md:55]: Yes, exactly.

**Comment by csomh** [comment]
> Since you pointed out so many high-volume projects - would a survey be an efficient way of gathering feedback?

Maybe, though I would probably find quick, one-on-one discussions or chats more useful. The reason for this is that surveys are not that good at finding the "unknown unknowns". A one-on-one discussion, being less orchestrated, can surface many of these. 

We don't necessarily need to interview every single of these projects. Getting the same answers over and over again or not learning anything new from subsequent interviews could be a good sign to know when to stop.

**Comment by TomasTomecek** [comment]
> > Since you pointed out so many high-volume projects - would a survey be an efficient way of gathering feedback?
> 
> Maybe, though I would probably find quick, one-on-one discussions or chats more useful. The reason for this is that surveys are not that good at finding the "unknown unknowns". A one-on-one discussion, being less orchestrated, can surface many of these.
> 
> We don't necessarily need to interview every single of these projects. Getting the same answers over and over again or not learning anything new from subsequent interviews could be a good sign to know when to stop.

Agreed. My concern was that such conversations would be really time-consuming but you're right that we can stop with those any time and reach out to the rest via an email.

Thanks again for this write-up, seems to me this is exactly what we need at this point to understand how can we serve our users better and progress further with the project.

**Pull request merged**